### PR TITLE
Password check

### PR DESCRIPTION
**Pull request by lubankp**
@lubankp added homework password check

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @lubankp

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @lubankp

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @lubankp

**Comment by ziobron** [comment]
⏱ 3 XP extra for @lubankp for on-time delivery

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,113 @@
...
+        result = "Password needs to have at least one special character";
+        break;
+    case ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter :
+        result = "Password needs to have at least one uppercase letter";
+        break;
+    case ErrorCode::PasswordsDoNotMatch :
+        result = "Passwords do not match";
+        break;
+    default :
+    result = "Unknown error";
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:29]: space

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,113 @@
...
+    }
+
+    return result;
+
+}
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:34]: Add new line between functions

File: homework/password-check/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,113 @@
...
+bool doPasswordsMatch(std::string password,std::string repeatedPassword){
+
+    if (password == repeatedPassword){
+        return true;
+    }else{
+        return false;
+    }
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:41]: just return lsh == rhs

File: homework/password-check/validation.cpp [hunk 4]
```diff
@@ -1,2 +1,113 @@
...
+        result = "Passwords do not match";
+        break;
+    default :
+    result = "Unknown error";
+    }
+
+    return result;
+
+}
+bool doPasswordsMatch(std::string password,std::string repeatedPassword){
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:35]: space!

File: homework/password-check/validation.cpp (hunk 3 above)
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:41]: Add space after `if`, `for`, `while` `function` etc...

Add space between operator like `+` `-` `/` `:` `&&` etc..

File: homework/password-check/validation.cpp (hunk 4 above)
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:35]: Pass by const refrence. Try always avoid copy.

File: homework/password-check/validation.cpp [hunk 5]
```diff
@@ -1,2 +1,113 @@
...
+    }else{
+        return false;
+    }
+        
+}
+ErrorCode checkPasswordRules(std::string password){
+    
+    if (password.length() < 9){
+        return ErrorCode::PasswordNeedsAtLeastNineCharacters;
+    }else{
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:48]: Add space after `if`, `for`, `while` `function` etc...

Add space between operator like `+` `-` `/` `:` `&&` etc..

File: homework/password-check/validation.cpp [hunk 6]
```diff
@@ -1,2 +1,113 @@
...
+        auto number_digits = 0;
+        for (auto i = 0; i < password.length(); i++){
+            if (std::isdigit(password[i])){  
+                number_digits++;            
+            }
+        }
+        if (number_digits == 0){
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:56]: Wrong indentation.

File: homework/password-check/validation.cpp [hunk 7]
```diff
@@ -1,2 +1,113 @@
...
+                        (password[i] >= 58 && password[i] <= 64) ||
+                        (password[i] >= 91 && password[i] <= 96) ||
+                        (password[i] > 123)){   
+                        number_specials++;            
+                    }
+                }
+                if (number_specials == 0){
+                return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+                }else{
+                return ErrorCode::Ok;
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:79]: When you use return inside `if` you don't need `if else` or `else`. Just write `if (sth) { return blabla; }`

File: homework/password-check/validation.cpp [hunk 8]
```diff
@@ -1,2 +1,113 @@
...
+
+    bool match = doPasswordsMatch(password,repeatedPassword);
+    ErrorCode error_code;
+    if(!match){
+        error_code =  ErrorCode::PasswordsDoNotMatch;
+    }else{
+
+        error_code = checkPasswordRules(password);
+    }
+    return error_code;
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:98]: When you use return inside `if` you don't need `if else` or `else`. Just write `if (sth) { return blabla; }`

**Review by nauka-programowania-MA — APPROVED** [review]
Please format your code

**Pull request closed without merging**